controllers/manilaapi_controller.go

```diff
@@ -638,30 +638,43 @@ func (r *ManilaAPIReconciler) reconcileNormal(ctx context.Context, instance *man
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ctrlResult, err := r.getSecret(ctx, helper, instance, instance.Spec.TransportURLSecret, &configVars)
+
+	hash, ctrlResult, err := ensureSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.TransportURLSecret},
+		[]string{
+			"transport_url",
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		manila.NormalDuration,
+	)
 	if err != nil {
 		return ctrlResult, err
 	}
 
+	// Add a prefix to the var name to avoid accidental collision with other non-secret
+	// vars. The secret names themselves will be unique.
+	configVars["secret-"+instance.Spec.TransportURLSecret] = env.SetValue(hash)
+
 	//
 	// check for required service secrets
 	//
 	for _, secretName := range instance.Spec.CustomServiceConfigSecrets {
-		ctrlResult, err = r.getSecret(ctx, helper, instance, secretName, &configVars)
+		ctrlResult, err = getConfigSecret(ctx, helper, &instance.Status.Conditions, secretName, instance.Namespace, &configVars)
 		if err != nil {
 			return ctrlResult, err
 		}
```

Comment:
Same question here about checking for `ctrlResult != ctrl.Result{}`

Reply:
same answer applies here

Reply:
Done